### PR TITLE
feat(create-eslint-config): implement setup wizard

### PR DESCRIPTION
**Pull request by hanna-skryl**
Partially addresses #47 

## Summary

- Interactive setup wizard for `@code-pushup/eslint-config`, triggered via `npm init @code-pushup/eslint-config` or imported programmatically
- Detects project state (deps, files, tsconfig, node version, existing eslint config) and pre-selects recommended configs
- Generates `eslint.config.mjs` (or `.js` for `type: "module"` projects) with preset inheritance
- Installs all required peer dependencies via the detected package manager
- Supports `--yes`, `--configs`, `--tsconfig`, `--node-version-source`, `--node-version`, and `--dry-run` CLI flags
- Full README with usage, options table, programmatic API, and available configs

> [!NOTE]
> Merging into existing `eslint.config.*` files is not yet implemented. The wizard prints a snippet for the user to paste manually. Handling `defineConfig` and `tseslint.config` wrappers will ship in a follow-up. Relevant code is marked with TODO comments.

### CODE REVIEW

File: packages/create-eslint-config/src/lib/wizard.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  const normalized = { ...options, targetDir };
+
+  const snapshot = await snapshotProject(targetDir);
+  const configs = await promptConfigSelection(normalized, snapshot);
+  const followUps = await collectFollowUps(configs, normalized, snapshot);
+
+  const existingConfig = await detectExistingEslintConfig(snapshot);
+  if (existingConfig?.format === 'cjs') {
+    throw new Error('Only ESLint configs in ESM format are supported.');
+  }
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/wizard.ts:28]: We can terminate early if the config is in CommonJS, no need to collect prompts in that case.

```suggestion
  const snapshot = await snapshotProject(targetDir);

  const existingConfig = await detectExistingEslintConfig(snapshot);
  if (existingConfig?.format === 'cjs') {
    throw new Error('Only ESLint configs in ESM format are supported.');
  }

  const configs = await promptConfigSelection(normalized, snapshot);
  const followUps = await collectFollowUps(configs, normalized, snapshot);
```

File: packages/create-eslint-config/src/lib/config-registry.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    isRecommended: snapshot =>
+      snapshot.files.has('tsconfig.json') || snapshot.allDeps.has('typescript'),
+  },
+  {
+    slug: 'node',
+    title: 'Node.js',
+    extends: 'javascript',
+    peerDeps: [{ name: 'eslint-plugin-n', version: '>=17.0.0' }],
+    isRecommended: snapshot => hasAnyDep(snapshot, BACKEND_FRAMEWORKS),
+  },
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/config-registry.ts:71]: The list of configs and peer dependencies needs to be dynamic. Otherwise, it'll be difficult to keep them in sync. They change fairly frequently in `@code-pushup/eslint-config`, and updating the setup wizard's constants each time would be cumbersome and error-prone. The `eslint-config` docs are auto-generated for the same reason.

I see you've added some dynamic tests for sanity checks, but they don't provide full coverage (e.g., peer deps per config), and still require manual intervention.

I believe it's necessary for `@code-pushup/create-eslint-config` to have `@code-pushup/eslint-config` as a dependency, so that it can infer configs and their dependencies automatically. Some of this code can probably be reused from the docs scripts.

File: packages/create-eslint-config/src/lib/codegen.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  ];
+}
+
+function nodeVersionExpression(node: NodeSetup): string {
+  if (node.source === 'node-version') {
+    return `${readFileSyncCall('.node-version')}.trim()`;
+  }
+  if (node.source === 'engines') {
+    return `JSON.parse(${readFileSyncCall('package.json')}).engines.node`;
+  }
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/codegen.ts]: If `engines.node` from `package.json` is used, then the Node version doesn't need to be set in the config (see [docs](https://github.com/eslint-community/eslint-plugin-n#configured-nodejs-version-range)).

File: packages/create-eslint-config/src/lib/utils.spec.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      devDependencies: { b: '2' },
+      peerDependencies: { c: '3' },
+    });
+    expect([...deps].toSorted()).toEqual(['a', 'b', 'c']);
+  });
+
+  it('should return an empty set for null input', () => {
+    expect(collectAllDeps(null).size).toBe(0);
+  });
+});
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/utils.spec.ts:72]: I'd add a test case for duplicate packages. It's quite common to have the same package in `devDependencies` (for local development) and `peerDependencies` (for publishing).

File: packages/create-eslint-config/src/lib/utils.spec.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    await writeFile(filePath, JSON.stringify({ name: 'foo' }));
+    await expect(readJsonFile(filePath)).resolves.toEqual({ name: 'foo' });
+  });
+
+  test('should return null for a missing file', async ({ tmp }) => {
+    await expect(
+      readJsonFile(path.join(tmp, 'missing.json')),
+    ).resolves.toBeNull();
+  });
+});
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/utils.spec.ts:29]: I'd add a test case for an existing file with invalid JSON.

File: packages/create-eslint-config/src/lib/prompts.ts [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+          ],
+          default: defaultSource,
+        }));
+
+  const detectedVersion =
+    source === NODE_VERSION_FILE
+      ? nodeInfo.file
+      : source === 'engines'
+        ? nodeInfo.engines
+        : undefined;
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/prompts.ts]: I don't understand why `'node-version'` is referenced indirectly via a `NODE_VERSION_FILE` constant, while `'engines'` and `'manual'` are referenced directly. 😕

File: packages/create-eslint-config/src/lib/prompts.ts [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+  const defaultVersion = `>=${process.versions.node}`;
+
+  const version =
+    options.nodeVersion ??
+    detectedVersion ??
+    (options.yes
+      ? defaultVersion
+      : await inputPrompt({
+          message: 'Node version range (e.g. >=20.0.0):',
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/prompts.ts]: Maybe referencing the current LTS is a better example:

```suggestion
          message: 'Node version range (e.g. >=24):',
```

File: packages/create-eslint-config/src/lib/prompts.ts [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+        }));
+
+  const detectedVersion =
+    source === NODE_VERSION_FILE
+      ? nodeInfo.file
+      : source === 'engines'
+        ? nodeInfo.engines
+        : undefined;
+
+  const defaultVersion = `>=${process.versions.node}`;
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/prompts.ts]: Nice idea to use the current Node version. 👍 

Though I'm not sure it's a good idea to allow major version updates by default.

```suggestion
  const defaultVersion = `^${process.versions.node}`;
```

We could also set the current Node version without a range:


```suggestion
  const defaultVersion = process.versions.node;
```

File: packages/create-eslint-config/src/lib/prompts.ts [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+  const version =
+    options.nodeVersion ??
+    detectedVersion ??
+    (options.yes
+      ? defaultVersion
+      : await inputPrompt({
+          message: 'Node version range (e.g. >=20.0.0):',
+          default: defaultVersion,
+          validate: value =>
+            value.trim() ? true : 'Node version range is required.',
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/prompts.ts]: We can use npm's own [`semver`](https://github.com/npm/node-semver#readme) parser to validate the syntax.

File: packages/create-eslint-config/src/cli.ts [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+  if (argv.dryRun) {
+    logInfo('Dry run — no files written.');
+  } else {
+    await result.flush();
+    const manager = await detectPackageManager(targetDir);
+    logInfo(`Installing dependencies with ${manager}...`, '');
+    try {
+      await installDependencies(targetDir);
+    } catch (error) {
```
Comment (matejchalk) [inline packages/create-eslint-config/src/cli.ts:78]: The `detectPackageManager` function is called twice - once here and once in `installDependencies`.

File: packages/create-eslint-config/src/test-setup.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+ * Vitest test with a per-test `tmp` fixture: a fresh temporary directory
+ * that is automatically removed when the test settles.
+ */
+export const test = baseTest.extend<{ tmp: string }>({
+  tmp: async ({ task: _task }, use) => {
+    const dir = await mkdtemp(path.join(tmpdir(), 'create-eslint-config-'));
+    await use(dir);
+    await rm(dir, { recursive: true, force: true });
+  },
+});
```
Comment (matejchalk) [inline packages/create-eslint-config/src/test-setup.ts:16]: I didn't know Vitest had this feature, very cool. 👍 

But why not use the recommended [builder pattern](https://vitest.dev/guide/test-context.html#builder-pattern)? 🤔

File: packages/create-eslint-config/package.json [hunk 12]
```diff
@@ -1,11 +1,33 @@
...
+    "setup-wizard",
+    "code-pushup"
+  ],
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/code-pushup/eslint-config/issues"
+  },
+  "homepage": "https://github.com/code-pushup/eslint-config/tree/main/packages/create-eslint-config#readme",
+  "dependencies": {
+    "@inquirer/prompts": "^8.4.1",
```
Comment (matejchalk) [inline packages/create-eslint-config/package.json:29]: Although this doesn't happen in CI, I get lint errors locally, even after removing `node_modules` and re-installing.

```
  29:5  error  The version specifier does not contain the installed version of "@inquirer/prompts" package: 7.10.1  @nx/dependency-checks
```

I think it's because `@inquirer/prompts` isn't installed in the monorepo root.

File: package.json [hunk 13]
```diff
@@ -2,6 +2,9 @@
   "name": "@code-pushup/eslint-config-workspace",
   "private": true,
   "type": "module",
+  "workspaces": [
+    "packages/*"
+  ],
```
Comment (matejchalk) [inline package.json]: I wouldn't recommend using npm workspaces. It's simpler to install all dependencies at the monorepo level, which (aside from `@inquirer/prompts` from my previous comment) is how the monorepo is set up. We tried npm workspaces in Code PushUp CLI, but had a lot of problems with stale builds.

File: packages/create-eslint-config/package.json [hunk 14]
```diff
@@ -1,11 +1,33 @@
...
+    "yargs": "^18.0.0",
+    "vitest": "^3.2.4"
```
Comment (matejchalk) [inline packages/create-eslint-config/package.json]: Vitest isn't a production dependency. I'm guessing this was motivated by the `@nx/dependency-checks` lint rule. It's caused by the `testing.ts` file, which needs to be excluded from the `production` named input in `nx.json`. Adding `"!{projectRoot}/src/testing.ts"` solves the problem.

Having said that, `src/testing.ts` doesn't seem like a very standard location. I've come across `test-setup.ts` (used by Jest). Or we could place it under an excluded folder like `mocks` or `test`.

File: packages/create-eslint-config/src/test-setup.ts (hunk 11 above)
Comment (hanna-skryl) [reply packages/create-eslint-config/src/test-setup.ts:16]: The builder pattern landed in Vitest 4.1.0. On v3 the only fixture API is the object syntax. The Vitest major bump is worth its own PR, so let's switch to the builder pattern there.

File: packages/create-eslint-config/src/lib/config-registry.spec.ts [hunk 15]
```diff
@@ -1,37 +1,16 @@
...
-const eslintConfigPeerDeps: Record<string, string> =
-  eslintConfigPackageJson.peerDependencies ?? {};
 
 describe('CONFIG_REGISTRY', () => {
-  it('should contain 13 entries matching the documented config list', () => {
-    expect(CONFIG_REGISTRY).toHaveLength(13);
+  it('should enumerate every config exported by eslint-config', () => {
+    expect(ALL_SLUGS).toEqual(expect.arrayContaining(['javascript', 'react']));
+    expect(CONFIG_REGISTRY.length).toBeGreaterThan(0);
   });
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/config-registry.spec.ts]: The test title overstates what it's validating. Something like "should enumerate configs exported by eslint-config" would be more accurate.

File: packages/create-eslint-config/src/lib/config-registry.ts [hunk 16]
```diff
@@ -30,178 +26,132 @@ function hasAnyFile(snapshot: ProjectSnapshot, pattern: RegExp): boolean {
...
+  javascript: {
     title: 'JavaScript (default)',
-    peerDeps: [],
     isRecommended: () => true,
   },
-  {
-    slug: 'typescript',
+  typescript: {
     title: 'TypeScript (strict)',
     extends: 'javascript',
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/config-registry.ts:56]: The relationships between configs can also be dynamically computed. This is how the docs script does it:

https://github.com/code-pushup/eslint-config/blob/aae8ff6d9243b43f8f1b6caf1c0ca9bd9eed76a8/packages/eslint-config/scripts/docs.js#L260-L271

File: packages/create-eslint-config/src/lib/config-registry.ts [hunk 17]
```diff
@@ -30,178 +26,132 @@ function hasAnyFile(snapshot: ProjectSnapshot, pattern: RegExp): boolean {
...
+  ...UNKNOWN_SLUGS,
+].map(slug => {
+  const meta = CONFIG_METADATA[slug];
+  return {
+    slug,
+    title: meta?.title ?? slugToTitle(slug),
+    ...(meta?.extends && { extends: meta.extends }),
+    isRecommended: meta?.isRecommended ?? (() => false),
+  };
+});
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/config-registry.ts]: This is an interesting approach; it'll handle missing config metadata gracefully. However, I'd like all future configs to be integrated with the setup wizard from day one. So I think a better approach would be to add a unit test that checks if all configs from `listAvailableConfigs` are defined in the `CONFIG_METADATA`.

File: packages/create-eslint-config/src/lib/eslint-config-source.ts [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+  const entries = await readdir(CONFIGS_DIR);
+  return entries
+    .filter(name => name.endsWith('.js'))
+    .map(name => name.replace(/\.js$/, ''));
+}
+
+/** Reads a config file's source text (used to introspect imports without loading the module). */
+export async function loadConfigSource(slug: string): Promise<string> {
+  return readFile(path.join(CONFIGS_DIR, `${slug}.js`), 'utf8');
+}
```
Comment (matejchalk) [inline packages/create-eslint-config/src/lib/eslint-config-source.ts]: I don't see how this approach can work in a published package. The relative path from `@code-pushup/create-eslint-config` files to `@code-pushup/eslint-config` won't match our local development folder structure. We need to `import ... from '@code-pushup/eslint-config/...'` so that user installations are resolved to the correct `node_modules` location. For the purposes of our local development, this can be replicated by a tsconfig path alias.

The more I think about it, the more it makes sense to add this derived metadata as an exported path in `@code-pushup/eslint-config/metadata.js` (or some other sub-path defined with `exports` in its `package.json`). This would provide better encapsulation and be reused by the docs script as well. 🤔 

I'm fine with doing this in another PR, though. We can keep the list of configs static in this PR (i.e., what you had originally), and make it dynamic in a follow-up, because it entails more complexity than I initially realized.

File: packages/create-eslint-config/eslint.config.js [hunk 19]
```diff
@@ -3,4 +3,24 @@
...
+    },
+  },
+  {
+    files: ['**/package.json'],
+    rules: {
+      // sibling install target, not an imported module
+      '@nx/dependency-checks': [
+        'error',
+        { ignoredDependencies: ['@code-pushup/eslint-config'] },
+      ],
```
Comment (matejchalk) [inline packages/create-eslint-config/eslint.config.js]: I think it does actually need to be imported, see previous comment.

File: packages/create-eslint-config/src/lib/eslint-config-source.ts (hunk 18 above)
Comment (hanna-skryl) [reply packages/create-eslint-config/src/lib/eslint-config-source.ts]: Agreed, let's defer. My preference would be to implement the extension of detected existing ESLint configs next, and revisit the dynamic registry once that lands.

File: packages/create-eslint-config/src/lib/config-registry.ts (hunk 17 above)
Comment (hanna-skryl) [reply packages/create-eslint-config/src/lib/config-registry.ts]: You're right, the dynamic version had a gap. The revert to static closes it, but it's worth revisiting when configs go dynamic again.

File: packages/create-eslint-config/src/lib/config-registry.ts (hunk 16 above)
Comment (hanna-skryl) [reply packages/create-eslint-config/src/lib/config-registry.ts:56]: I've added a TODO so we don't lose sight of it in the follow-up.

File: packages/create-eslint-config/eslint.config.js [hunk 20]
```diff
@@ -3,24 +3,11 @@
...
-    files: ['**/package.json'],
-    rules: {
-      // sibling install target, not an imported module
-      '@nx/dependency-checks': [
-        'error',
-        { ignoredDependencies: ['@code-pushup/eslint-config'] },
-      ],
-    },
-  },
-);
```
Comment (matejchalk) [inline packages/create-eslint-config/eslint.config.js]: The `@nx/dependency-checks` rule should stay in place (without the ignored dependencies). 🙏

**Review by matejchalk — CHANGES_REQUESTED** [review]

File: packages/create-eslint-config/eslint.config.js (hunk 19 above)
Comment (matejchalk) [inline packages/create-eslint-config/eslint.config.js]:
```suggestion
      '@nx/dependency-checks': 'error',
```

**Review by matejchalk — APPROVED** [review]
